test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts

```diff
@@ -43,6 +43,10 @@ describe(
 				editorPage = new EditorPage( page, { target: features.siteType } );
 			} );
 
+			afterAll( async () => {
+				await page.close();
+			} );
+
```

Comment:
**Question**: Do we need this here? Aren't all the pages closed up by our custom jest environment? AFAIK, we usually don't need to close up pages manually during tests.

Reply:
And actually, I think that may mess with capturing screenshots and videos, because we iterate through open pages on teardown and grab the screenshots and videos. But if the page is already closed, the teardown logic won't find it!

Reply:
Perfect catch! I've just realized that the `after` hook is fired even if the block fails, making it the wrong spot to put the page closing at because it indeed would prevent the artifacts from being created. 💥 In the case of suites like these, though, where there are multiple `describe` blocks, I'd still close the page at the end of every block, only in a dedicated step instead of the `after` hook. Otherwise, the recordings will not be stopped even for successful blocks and will be uploaded along with failed ones. For example, this will upload three sets of artifacts, of which only the last one is relevant:

```js
// foo.spec.js

describe({
  // opens page #1, succeeds
})
describe({
  // page #1 is open
  // opens page #2, succeeds
})
describe({
  // page #1 is open
  // page #2 is open
  // opens page #3, fails
})
```


This is true for our current env setup, so alternatively, maybe we could make some changes not to create artifacts for successful blocks? Let me know what you think, @dpasque! 

---

Accidentally, while testing if what I've described above is true, I've uncovered a bug in our env setup that causes each artifact (trace/recording/image/etc.) to overwrite the previous one. It happens because we're creating the artifact name [outside of the context→page iteration blocks](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/jest-playwright-config/environment.ts#L155-L180), which causes each context & page have the `0` index. I'll make a fix in another PR!

Reply:
Draft here: https://github.com/Automattic/wp-calypso/pull/71623

Reply:
**comment**: A counterpoint is that we may have a situation where the previous top-level `describe` blocks may have passed but it inadvertently put the webapp in a condition that leads to an immediate failure in the next top-level `describe` block.

In the hypothetical spec with three `describe` blocks above, it is possible that an overlay appears between the last step of the second block and the beginning of the third block on the same page.

Granted, it is little bit of a cherry-picked scenario, I will admit that!